packages/xo-server/src/xapi/mixins/patching.js

```diff
@@ -464,4 +465,102 @@ export default {
     // host-by-host install
     throw new Error('non pool-wide install not implemented')
   },
+
+  async rollingPoolUpdate() {
+    let hosts = filter(this.objects.all, { $type: 'host' })
+    await Promise.all(hosts.map(host => this.call('host.assert_can_evacuate', host.$ref)))
```

Comment:
You can also do this with records :slightly_smiling_face: 

```suggestion
    await Promise.all(hosts.map(host => host.$call('assert_can_evacuate')))
```